spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java

```diff
@@ -0,0 +1,283 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.debezium.inbound;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.engine.DebeziumEngine.Builder;
+import io.debezium.engine.DebeziumEngine.ChangeConsumer;
+import io.debezium.engine.DebeziumEngine.RecordCommitter;
+import io.debezium.engine.Header;
+import io.debezium.engine.format.SerializationFormat;
+
+import org.springframework.integration.debezium.inbound.support.DebeziumHeaders;
+import org.springframework.integration.debezium.inbound.support.DefaultDebeziumHeaderMapper;
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.integration.support.AbstractIntegrationMessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.HeaderMapper;
+import org.springframework.scheduling.concurrent.CustomizableThreadFactory;
+import org.springframework.util.Assert;
+
+/**
+ * Debezium Change Event Channel Adapter.
+ *
+ * @author Christian Tzolov
+ * @since 6.2
+ */
+public class DebeziumMessageProducer extends MessageProducerSupport {
+
+	private final DebeziumEngine.Builder<ChangeEvent<byte[], byte[]>> debeziumEngineBuilder;
+
+	private DebeziumEngine<ChangeEvent<byte[], byte[]>> debeziumEngine;
+
+	/**
+	 * Debezium Engine is designed to be submitted to an {@link Executor} or {@link ExecutorService} for execution by a
+	 * single thread. By default a single-threaded ExecutorService instance is provided configured with a
+	 * {@link CustomizableThreadFactory} and a `debezium-` thread prefix.
+	 */
+	private ExecutorService executorService;
+
+	private CountDownLatch latch = new CountDownLatch(0);
+
+	private String contentType = "application/json";
+
+	private HeaderMapper<List<Header<Object>>> headerMapper = new DefaultDebeziumHeaderMapper();
+
+	private boolean enableEmptyPayload = false;
+
+	private boolean enableBatch = false;
+
+	private ThreadFactory threadFactory;
+
+	/**
+	 * Create new Debezium message producer inbound channel adapter.
+	 * @param debeziumBuilder - pre-configured Debezium Engine Builder instance.
+	 */
+	public DebeziumMessageProducer(Builder<ChangeEvent<byte[], byte[]>> debeziumBuilder) {
+		Assert.notNull(debeziumBuilder, "The Debezium Engine Builder is null!");
```

Comment:
Don't yell to end-users.
Just simple ` "'debeziumBuilder' must not be null"` is enough.